eeprom.c

```diff
@@ -157,25 +157,101 @@ int start_programming(llio_t *self, u32 start_address, int fsize) {
     return 0;
 }
 
-int eeprom_write(llio_t *self, char *bitfile_name, u32 start_address, int fix_boot_flag) {
+int eeprom_write(llio_t *self, char *bitfile_name, u32 start_address, int fix_boot_flag, int sha256_check_flag) {
     board_t *board = self->board;
-    int bytesread, i;
+    int bytesread, i, j;
     u32 eeprom_addr;
     char part_name[32];
     struct stat file_stat;
     FILE *fp;
     struct timeval tv1, tv2;
+    char sha256str[65];
+    char sha256file_name[262];
+    unsigned char sha256in[32];
+    unsigned char sha256bitfile[32];
+    SHA256_CTX sha256ctx;
 
-    if (stat(bitfile_name, &file_stat) != 0) {
-        printf("Can't find file %s\n", bitfile_name);
-        return -1;
-    }
+    if (sha256_check_flag) {
+        printf("\nFPGA configuration bitfile integrity verification:\n");
+        strcpy(sha256file_name, bitfile_name);
+        strcat(sha256file_name, ".sha256");
 
-    fp = fopen(bitfile_name, "rb");
-    if (fp == NULL) {
-        printf("Can't open file %s: %s\n", bitfile_name, strerror(errno));
-        return -1;
+        if (stat(sha256file_name, &file_stat) != 0) {
+            printf("Can't find checksum file '%s'\n", sha256file_name);
+            return -1;
+        }
```

Comment:
can this code be shared with the _other_ sha256 code added by your other PR?  For instance, a new function `bool verify_sha256(const char *filename)`  Otherwise, there are two copies of the code, which each may have different bugs.